sim/midas/src/main/scala/midas/widgets/Bridge.scala

```diff
@@ -47,143 +43,61 @@ abstract class BridgeModuleImp[HostPortType <: TokenizedRecord]
 
 }
 
-trait Bridge[HPType <: TokenizedRecord, WidgetType <: BridgeModule[HPType]] {
-  self: BaseModule =>
+/**
+  * Target-side base-mixin for defining a Bridge. We expect this to be extended in two ways:
+  * 1) By mixing this into a target module's class, as done in the original
+  *    implementations. See [[Bridge]].
+  * 2) By implementing the abstract members in an annotator object. This permits marking modules 
+  *    in code bases that do not depend on Golden Gate as bridges.
+  */
+trait Bridgeable[HPType <: Record with HasChannels, WidgetType <: BridgeModule[HPType]] {
   def constructorArg: Option[_ <: AnyRef]
   def bridgeIO: HPType
+  def target: ModuleTarget
 
   def generateAnnotations(): Unit = {
```

Comment:
There is no strong reason, it was just the first thing that occurred to me. I think in some cases you may wish to do some intermediate processing that you could wrap up. 

This is basically the use case: 
https://github.com/firesim/firesim/blob/5b5d8a4038210b15c520718e45d389d62050fb24/sim/midas/src/main/scala/midas/widgets/ClockMux.scala#L105

If i understand your second point correctly, there's nothing stopping me from writing a generic AnnotateAsBridge method, that takes the arguments you suggest and does the right thing.  It'd probably still be good to have dedicated methods (i.e., type checked) for labelling specific Modules as specific bridges, but this could just be a parameterization of the above? I'll try playing around with that.

Reply:
I've decided to drop this "feature" until there's a motivating use case that i'd like to deploy immediately.

Reply:
If the "inherit from bridgeable" were replaced with "call a utility," the linked code would look like the following:

```scala
// class BridgeableClockMux deleted entirely!

object BridgeableClockMux {
  def apply(mod: BaseModule, clockA: Clock, clockB: Clock, clockOut: Clock, sel: Bool,
            implStyle: ClockMuxImplStyle = MuxStyle.Mutex(3)): Unit = {
    val io = new ClockMuxHostIO(clockA, clockB, clockOut, sel)
    SingletonContainingBridgeableUtilitiesOrWhereverTheCodeLives.generateAnnotations(mod, io, implStyle)
  }
}
```

Basically, the crux of the matter is that there is some necessary "mutate global state at a distance" code to generate bridge-specifying annotations. I think of `(target, host_io, constructor_arg)` as the conceptual parameters to that code, and my suggestion is basically that it's easier (and equally easy to implement type-safe interfaces and to allow pre-processing) to just pass them as parameters rather than making an object whose lifetime is intrinsically tied to exactly one invocation of that "mutate global state at a distance" code.

Reply:
No i understand and completely agree. I prototyped something based on your suggestion which worked. Ultimately i decided it'd be more sensible to add that implementation when it was actually needed -- the scope of this PR was already too large.

Reply:
If i'm being honest, that's another way of saying i can't be fucked to write tests for it right now, because i have other stuff i need to do.